### ISSUE TITLE
Premature success event from observables in NgxIndexedDBService

### ISSUE THREAD

**Issue by marcindz88** [assuncaocharles/ngx-indexed-db#402]
Hi,
I have used the library to add or update a large amount of data in indexedDB, and I have observed that the observable for `update` emits just as the update has succeeded, but I think it should emit only once the transaction created with it completes. 

Currently, the observable emits that a row has been updated, however the real write to memory happens a few seconds later, which may be a problem for subsequent operations or the ability to show the user a loader or some information that the data is still saving to the DB. 

Also, I think it would be nice to include a possibility to execute operations on multiple tables in a bulk transaction, but this is kind of a separate issue.

**Comment by aparzi** [comment]
Hi @marcindz88,
For large amount of data there are available bulk operation, in particolary bulk update.

For execute operations in multiple tables, you can open an issue and describe your problem and possible solution or if you want contrbuition feel free to do so

**Comment by marcindz88** [comment]
Bulking will not change anything, library should listen for complete event on transaction object https://developer.mozilla.org/en-US/docs/Web/API/IDBDatabase/transaction#examples. Otherwise we inform users that update has finished where in fact transaction has not been finalized.

**Comment by aparzi** [comment]
Sure, i will look into the problem as soon as i can. thanks for the feedback.

**Comment by JonataB** [comment]
I take care of It
Just doing resolve on transaction complete and check differences
Also prevent entire transaction abort if any item fails

**Comment by aparzi** [comment]
@JonataB you are welcome.
when you have made the changes please open the PR and I will check it. Thank you very much for the support

**Comment by JonataB** [comment]
Simple performance test repeated 3 times have following result:
bukl add on 200.000 items without errors 50175.80000001192 milliseconds
bukl add on 200.000 items with errors (100.000) 31137.80000001192 milliseconds

I used the items length where you are familiar with (200.000)
Every time best peformance is when we have error obviously

**Comment by JonataB** [comment]
I am taking care of this of ok for @aparzi
